### DISCUSSION TITLE
QuarkChain L2 testnet Transaction 一直 pending

### DISCUSSION THREAD

**Discussion by linghuccc** [IntensiveCoLearning/Web3-URL#213] [Q&A] [answered]
钱包地址：0x4d395feC86e88b85b087f58B16972De58FB9d920

我昨天发起了一个 QKC transfer request，但是一直没通过，我就尝试通过加 gas 的方式加速或者取消（max base fee 加到 100, priority fee 加到 50），但是发现没有反应。
![5](https://github.com/user-attachments/assets/e4cb09f8-2963-4c99-aab9-3419abcc43d9)
![6](https://github.com/user-attachments/assets/32d15d39-a549-44e3-8916-c084988e37bf)

于是我写了一个脚本，去查看我的地址是否有 pending transaction，脚本给出的结论是没有。
![3](https://github.com/user-attachments/assets/427718fa-acc1-4de2-882f-a05db3cde806)

所以我就做了 swap，结果一样 stuck 了。于是我决定等一晚，结果今天去看，在 MetaMask 里面还是没有变化。同样通过加 gas 的方式仍然无法加速成功或者取消成功。脚本给出的结论依然是没有 pending transaction。于是我又发起了一个 transfer request，结果还是一样 stuck。
![1](https://github.com/user-attachments/assets/eb982602-ade6-4deb-83ee-d53832732574)

而且当我尝试检查昨天的 transfer transaction，却查不到任何信息。
![4](https://github.com/user-attachments/assets/23e4f26a-c9a3-445d-80ae-9b47c8ff2714)
![2](https://github.com/user-attachments/assets/20a0d751-06c9-4c55-9ad7-88bafa6b5c70)

所以这几个 transaction 到底是 pending 还是什么情况？我应该怎么办？谢谢。

**Comment by qzhodl** [comment, marked as answer]
看看是不是nonce不对导致的。
刚用 cast nonce 0x4d395feC86e88b85b087f58B16972De58FB9d920 -r https://rpc.testnet.l2.quarkchain.io:8545 查了一下，应该是 9 才对

**Reply by linghuccc** [reply]
> 看看是不是nonce不对导致的。 刚用 cast nonce 0x4d395feC86e88b85b087f58B16972De58FB9d920 -r https://rpc.testnet.l2.quarkchain.io:8545 查了一下，应该是 9 才对

nonce 9 是最近的一次成功 transaction，另外 3 个 transaction 的 nonce 按照顺序排列，看起来并没有问题。
![nonce](https://github.com/user-attachments/assets/80725763-74a2-4f65-87f5-bf694527ec46)

问题是这 3 个 transaction stuck 在这里，现在无法继续。

**Reply by qzhodl** [reply]
我看浏览器里面你最新的一笔交易的nonce是8：https://explorer.testnet.l2.quarkchain.io/address/0x4d395feC86e88b85b087f58B16972De58FB9d920

**Reply by linghuccc** [reply]
> 我看浏览器里面你最新的一笔交易的nonce是8：https://explorer.testnet.l2.quarkchain.io/address/0x4d395feC86e88b85b087f58B16972De58FB9d920

是的，我刚才图上显示成功的 nonce 为 9 的 transaction，在浏览器里面查不到。

**Reply by linghuccc** [reply]
现在已经能够继续了，谢谢

**Comment by qzhodl** [comment]
metamask里面可以通过设置，来手动设置nonce。settings => advanced => customize transaction nonce
